ArduCopter/inertia.cpp

```diff
@@ -3,28 +3,64 @@
 // read_inertia - read inertia in from accelerometers
 void Copter::read_inertia()
 {
-    // inertial altitude estimates
-    inertial_nav.update(G_Dt);
+    // pull position
+    if (!ahrs.get_origin(ekf_origin)) {
+        copter_state.has_ekf_update = false;
+        return;
+    }
+    copter_state.has_ekf_update = true;
+
+    Location loc{};
+    if (!ahrs.get_position(loc)) {
+        copter_state.has_current_loc = false;
+        return;
+    }
+    current_loc.lng = loc.lng;
+    current_loc.lat = loc.lat;
+    copter_state.has_current_loc = true;
+
+    // Get XYZ position and velocity in NEU and cm
+    if (!ahrs.get_relative_position_NED_origin(current_pos)) {
+        copter_state.has_relative_pos = false;
+        return;
+    }
+    current_pos = current_pos * 100.0f;  // m to cm
+    current_pos.z = -current_pos.z;  // NED to NEU
+    copter_state.has_relative_pos = true;
 
-    // pull position from interial nav library
-    current_loc.lng = inertial_nav.get_longitude();
-    current_loc.lat = inertial_nav.get_latitude();
+    if (!ahrs.get_velocity_NED(current_vel)) {
+        copter_state.has_current_vel = false;
+        return;
+    }
+    current_vel = current_vel * 100.0f;  // m to cm
+    current_vel.z = -current_vel.z;  // NED to NEU
+    copter_state.has_current_vel = true;
 
     // exit immediately if we do not have an altitude estimate
-    if (!inertial_nav.get_filter_status().flags.vert_pos) {
+    if (!ahrs.get_filter_status(filt_status)) {
+        copter_state.has_filt_status = false;
+        return;
+    }
+    if (!filt_status.flags.vert_pos) {
+        copter_state.has_filt_status = false;
         return;
     }
+    copter_state.has_filt_status = true;
 
     // without home return alt above the EKF origin
     if (ap.home_state == HOME_UNSET) {
         // with inertial nav we can update the altitude and climb rate at 50hz
-        current_loc.alt = inertial_nav.get_altitude();
+        // get the D position relative to the local earth frame origin
+        float posD;
+        if (ahrs.get_relative_position_D_origin(posD)) {
+            current_loc.alt = -posD * 100;  // convert from m in NED to cm in NEU
+        }
     } else {
         // with inertial nav we can update the altitude and climb rate at 50hz
-        current_loc.alt = pv_alt_above_home(inertial_nav.get_altitude());
+        current_loc.alt = pv_alt_above_home(current_pos.z);
     }
 
     // set flags and get velocity
     current_loc.flags.relative_alt = true;
```

Comment:
We should remove this flag and change the code above to instead use the correct:

```
if (ap.home_state == HOME_UNSET) {
  current_loc.set_alt_cm(current_pos.z, ALT_FRAME_ABOVE_ORIGIN);
} else {
  current_loc.set_alt_cm(pv_alt_above_home(current_pos.z), ALT_FRAME_ABOVE_HOME);
}
```

Reply:
Agree and done !